docs/OpenSSL300Design.md

```diff
@@ -201,20 +201,23 @@ here are as follows:
 *   FIPS Provider: Implements a set of algorithms that are FIPS
     validated and are made available via the Core. This includes the
     following supporting services:
-    *   POST: Power On Self Test
-    *   KAT: Known Answer Tests
-    *   Integrity Check
+    *   POST: Power On Self Test that perform:
+        *   KAT: Known Answer Tests
+        *   Integrity Check
     *   Low Level Implementations: This is the set of components that
         actually implement the cryptographic primitives (to meet the
         FIPS-mandated self-contained requirement).
-*   Engines Provider: A shim that allows existing engines to work
-    when called via the Core.
 *   Legacy Provider: Provides implementations of older algorithms that
     will be exposed via EVP-level APIs.
 *   3rd Party Providers: Eventually, third-parties may provide their
     own providers. A third-party provider, like any other provider,
     implements a set of algorithms that will be accessible to
     applications and other providers via the Core.
+*   Null Provider: A provider that does nothing. This can be useful
+    for testing that the correct library context is used.
+*   Base Provider: A provider for serialization of keys. The FIPS
+    provider requires this since it does not contains methods to
+    load keys. The Base provider is also embedded in the core.
```

Comment:
"the core" may not be entirely clear. Do you mean "embedded in the default provider"?